builtin/bins/dkron-executor-shell/shell.go

```diff
@@ -96,18 +92,53 @@ func (s *Shell) ExecuteImpl(args *dktypes.ExecuteRequest, cb dkplugin.StatusHelp
 	stdin.Close()
 
 	log.Printf("shell: going to run %s", command)
+
+	jobTimeout := args.Config["timeout"]
+	var jt time.Duration
+
+	if jobTimeout != "" {
+		jt, err = time.ParseDuration(jobTimeout)
+		if err != nil {
+			return nil, errors.New("shell: Error parsing job timeout")
+		}
+	}
+
 	err = cmd.Start()
 	if err != nil {
 		return nil, err
 	}
 
-	// Warn if buffer is overritten
+	var jobTimeoutMessage string
+	var jobWasKilled bool
```

Comment:
The `jobWasKilled` value actually signifies something different. I suggest renaming it to `jobTimedOut`, so it is in line with `jobTimeoutMessage`.